### ISSUE TITLE
znh5md crashes when loading .traj file with dict in `atoms.info`

### ISSUE THREAD

**Issue by M-R-Schaefer** [zincware/ZnH5MD#140]
ASE build some times adds dicts to the atoms.info field and this is currently not handled by H5.

found by @lkkmpn

**Comment by lkkmpn** [comment]
Seems like the problem is that non-numeral datatypes in values in the `info` dictionary are not supported.

Minimum example:
```python
from ase.atoms import Atoms
import znh5md

atoms = Atoms(info={'hello': 'world'})
znh5md.write('test.h5', atoms)  % ← crash
```

Traceback:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/luuk/envs/str/lib/python3.11/site-packages/znh5md/znh5md.py", line 24, in write
    io.extend(images)
  File "/home/luuk/envs/str/lib/python3.11/site-packages/znh5md/io.py", line 199, in extend
    combined_data = fmt.combine_asedata(data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luuk/envs/str/lib/python3.11/site-packages/znh5md/format.py", line 214, in combine_asedata
    observables = _combine_dicts([x.observables for x in data])
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luuk/envs/str/lib/python3.11/site-packages/znh5md/format.py", line 257, in _combine_dicts
    combined[key] = concatenate_varying_shape_arrays(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luuk/envs/str/lib/python3.11/site-packages/znh5md/utils.py", line 32, in concatenate_varying_shape_arrays
    result[i, : x.shape[0], ...] = x
    ~~~~~~^^^^^^^^^^^^^^^^^^^^^^
ValueError: could not convert string to float: 'world'
```

**Comment by lkkmpn** [comment]
Actually caused by me running an older version 😅, was already fixed in 9b90446c

**Comment by PythonFZ** [comment]
Should also be covered by
https://github.com/zincware/ZnH5MD/blob/5340cc054b7edc72cd40e0ac38764c76717c6345/tests/test_data_types.py#L41-L47

I'll close this for now. If you encounter the issue again, please reopen.
